python/cuml/internals/base.pyx

```diff
@@ -423,6 +478,132 @@ def _determine_stateless_output_type(output_type, input_obj):
 
     # If we are using 'input', determine the the type from the input object
     if temp_output == 'input':
-        temp_output = cuml.common.input_utils.determine_array_type(input_obj)
+        temp_output = cuml.internals.input_utils.determine_array_type(input_obj)
 
     return temp_output
+
+
+class UniversalBase(Base):
```

Comment:
Can you provide more documentation on the purpose of this class? It is not immediately clear to me how it relates to the `Base` class.

Reply:
@viclafargue Do you want to handle this in your PR so you can describe the full dispatching mechanism in whatever depth is necessary?